File: sdks/php/examples/HelloWorldNode.php
<?php

declare(strict_types=1);

namespace Blok\Blok\Examples;

use Blok\Blok\Node\NodeHandler;
use Blok\Blok\Types\Context;

/**
 * HelloWorldNode greets the user with a configurable prefix.
 *
 * Config:
 *   - prefix (string, optional): Greeting prefix (default: "Hello")
 *
 * Body:
 *   - name (string, optional): Name to greet (default: "World")
 */
final class HelloWorldNode implements NodeHandler
{
    public function execute(Context $ctx, array $config): mixed
    {
        $name = $ctx->request->bodyStr('name') ?? 'World';
        $prefix = isset($config['prefix']) && is_string($config['prefix'])
            ? $config['prefix']
            : 'Hello';

        $message = sprintf('%s, %s!', $prefix, $name);

        // Store greeting in context for downstream nodes
        $ctx->setVar('greeting', $message);

        return [
            'message' => $message,
            'timestamp' => gmdate('Y-m-d\TH:i:s\Z'),
            'language' => 'php',
        ];
    }
}

File: sdks/ruby/examples/hello_world_node.rb
# frozen_string_literal: true

require_relative "../lib/blok"

# HelloWorldNode greets the user with a configurable prefix.
#
# Config:
#   - "prefix" (string, optional): Greeting prefix (default: "Hello")
#
# Request body:
#   - "name" (string, optional): Name to greet (default: "World")
#
# Output:
#   { "message" => "Hello, World!", "timestamp" => "...", "language" => "ruby" }
class HelloWorldNode < Blok::Node::NodeHandler
  def execute(ctx, config)
    name   = ctx.request.body_str("name") || "World"
    prefix = config["prefix"] || "Hello"

    message = "#{prefix}, #{name}!"

    # Store the greeting in context for downstream nodes
    ctx.set_var("greeting", message)

    {
      "message"   => message,
      "timestamp" => Time.now.utc.iso8601,
      "language"  => "ruby"
    }
  end
end

# ----- Boot the server if run directly -----
if __FILE__ == $PROGRAM_NAME
  registry = Blok::Server::RuntimeApp.registry
  registry.register("hello-world", HelloWorldNode.new)

  puts "Starting HelloWorldNode on port 8080..."
  Blok::Server::RuntimeApp.run!(port: 8080, bind: "0.0.0.0")
end
